app/services/mapping_migration_service.rb

```diff
@@ -0,0 +1,123 @@
+class MappingMigrationService
+  LEGACY_FIELDS_REGEX = /%(\S+?)%/
+  attr_reader :integration_name, :rtp_id
+
+  def call
+    upload_integrations.each do |integration|
+      @integration_name = integration.plugin_name.to_s
+      set_integration_template_dirs
+
+      if integration.uploaders.count > 1
```

Comment:
The only reason for this check here is that for our integrations with multiple uploaders we renamed their templates since they were not following a convention. Since we renamed them, the `Mapping` source isn't going to match the name of the `.template` file so we had to handle these cases differently